code/modules/spacepods/equipment.dm

```diff
@@ -0,0 +1,293 @@
+/obj/item/device/spacepod_equipment/weaponry/proc/fire_weapons()
+	if(my_atom.next_firetime > world.time)
+		to_chat(usr, "<span class='warning'>Your weapons are recharging.</span>")
+		return
+	var/turf/firstloc
+	var/turf/secondloc
+	if(!my_atom.equipment_system || !my_atom.equipment_system.weapon_system)
+		to_chat(usr, "<span class='warning'>Missing equipment or weapons.</span>")
+		my_atom.verbs -= text2path("[type]/proc/fire_weapons")
+		return
+	if(!my_atom.cell.use(shot_cost))
+		to_chat(usr, "<span class='warning'>Insufficient charge to fire the weapons</span>")
+		return
+	var/olddir
+	for(var/i = 0; i < shots_per; i++)
+		if(olddir != my_atom.dir)
+			switch(my_atom.dir)
+				if(NORTH)
+					firstloc = get_step(my_atom, NORTH)
+					secondloc = get_step(firstloc,EAST)
+				if(SOUTH)
+					firstloc = get_turf(my_atom)
+					secondloc = get_step(firstloc,EAST)
+				if(EAST)
+					firstloc = get_step(my_atom, EAST)
+					secondloc = get_step(firstloc,NORTH)
+				if(WEST)
+					firstloc = get_turf(my_atom)
+					secondloc = get_step(firstloc,NORTH)
+		olddir = dir
+		var/obj/item/projectile/projone = new projectile_type(firstloc)
+		var/obj/item/projectile/projtwo = new projectile_type(secondloc)
+		projone.starting = get_turf(my_atom)
+		projone.firer = usr
+		projone.def_zone = "chest"
+		projtwo.starting = get_turf(my_atom)
+		projtwo.firer = usr
+		projtwo.def_zone = "chest"
+		spawn()
```

Comment:
we don't allow ```spawn()```
Make the code in the spawn into a proc
```
/obj/item/device/spacepod_equipment/weaponry/proc/do_fire(obj/item/projectile/projone, obj/item/projectile/projtwo)
    playsound(src, fire_sound, 50, 1)
    projone.dumbfire(my_atom.dir)
    projtwo.dumbfire(my_atom.dir)
```

Then replace the spawn with:
```INVOKE_ASYNC(src, .proc/do_fire, projone, projtwo)```